src/main/java/com/refinedmods/refinedstorage/integration/jei/RecipeTransferGridError.java

```diff
@@ -0,0 +1,60 @@
+package com.refinedmods.refinedstorage.integration.jei;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import com.refinedmods.refinedstorage.util.TemporaryPortingUtils;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.recipe.transfer.IRecipeTransferError;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.IFormattableTextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecipeTransferGridError implements IRecipeTransferError {
+    private static final Color HIGHLIGHT_COLOR = new Color(1.0f, 0.0f, 0.0f, 0.4f);
+    private static final Color HIGHLIGHT_AUTOCRAFT_COLOR = new Color(0.0f, 0.0f, 1.0f, 0.4f);
+    private final List<IFormattableTextComponent> message = new ArrayList<>();
+    boolean craftMessage;
+    boolean missingMessage;
+    JEIIngredientTracker tracker;
```

Comment:
Please refactor your code so these attributes aren't needed, it's kinda ugly and makes for a confusing flow.